x/ratelimit/keeper/rate_limit.go

```diff
@@ -151,41 +187,95 @@ func (k Keeper) GetAllRateLimits(ctx sdk.Context) []types.RateLimit {
 	return allRateLimits
 }
 
+// Sets the sequence number of a packet that was just sent
+func (k Keeper) SetPendingSendPacket(ctx sdk.Context, channelId string, sequence uint64) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.PendingSendPacketPrefix)
+	key := types.GetPendingSendPacketKey(channelId, sequence)
+	store.Set(key, []byte{1})
+}
+
+// Remove a pending packet sequence number from the store
+// Used after the ack or timeout for a packet has been received
+func (k Keeper) RemovePendingSendPacket(ctx sdk.Context, channelId string, sequence uint64) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.PendingSendPacketPrefix)
+	key := types.GetPendingSendPacketKey(channelId, sequence)
+	store.Delete(key)
+}
+
+// Checks whether the packet sequence number is in the store - indicating that it was
+// sent during the current quota
+func (k Keeper) CheckPacketSentDuringCurrentQuota(ctx sdk.Context, channelId string, sequence uint64) bool {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.PendingSendPacketPrefix)
+	key := types.GetPendingSendPacketKey(channelId, sequence)
+	valueBz := store.Get(key)
+	found := len(valueBz) != 0
+	return found
+}
+
+// Get all pending packet sequence numbers
+func (k Keeper) GetAllPendingSendPackets(ctx sdk.Context) []string {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.PendingSendPacketPrefix)
+
+	iterator := store.Iterator(nil, nil)
+	defer iterator.Close()
+
+	pendingPackets := []string{}
+	for ; iterator.Valid(); iterator.Next() {
+		key := iterator.Key()
+
+		channelId := string(key[:types.PendingSendPacketChannelLength])
+		channelId = strings.TrimRight(channelId, "\x00") // removes null bytes from suffix
```

Comment:
Why are there null bytes?

Reply:
The length of the channel ID is fixed at 16 (related to above comment).

If you serialize a channelId of "channel-0", you get 
```
[99 104 97 110 110 101 108 45 48]
```

So then it gets zero-padded to 16 length
```
[99 104 97 110 110 101 108 45 48 0 0 0...]
```

And when you convert back to a string those zero's show up in the string
```
channel-0\x00\x00\0xx
```

Fwiw though, this is only used for export genesis - it's a pretty insignificant function

Reply:
Ah I see! Thanks for the explanation